tests/integration/components/hyper-table-v2-test.ts

```diff
@@ -157,46 +158,192 @@ module('Integration | Component | hyper-table-v2', function (hooks) {
       assert.dom('.upf-checkbox').exists({ count: 3 });
     });
 
-
     test('the selection column is present when the feature is enabled', async function (assert: Assert) {
       await render(hbs`<HyperTableV2 @handler={{this.handler}} @features={{hash selection=true}} />`);
 
       assert.dom('.selected-count').exists();
-    })
+    });
 
-    test('clicking the checkbox in the header of the selection column triggers the SelectAll', async function (assert: Assert) {
-      const handlerSpy = sinon.spy(this.handler);
-      await render(hbs`<HyperTableV2 @handler={{this.handler}} @features={{hash selection=true}} />`);
+    module('when clicking the checkbox in the header of the selection column with partial rows loaded', () => {
+      test('it triggers the handler SelectAll', async function (assert: Assert) {
+        const toggleSelectAllStub = sinon.stub(this.handler, 'toggleSelectAll');
+        await render(hbs`<HyperTableV2 @handler={{this.handler}} @features={{hash selection=true}} />`);
 
-      assert.deepEqual(this.handler.selection, []);
-      assert.dom('.hypertable__column.hypertable__column--selection header .upf-checkbox').exists();
+        assert.dom('.hypertable__column.hypertable__column--selection header .upf-checkbox').exists();
+        await click('.hypertable__column.hypertable__column--selection header .upf-checkbox');
 
-      await click('.hypertable__column.hypertable__column--selection header .upf-checkbox');
-      // @ts-ignore
-      assert.ok(handlerSpy.toggleSelectAll.calledOnceWithExactly(true));
-      assert.equal(this.handler.selection, 'all');
-      assert.dom('.selected-count').hasText(this.handler.rowsMeta.total.toString());
+        assert.ok(toggleSelectAllStub.calledOnceWithExactly(true));
+      });
+
+      test('it updates the selected count with correct number', async function (assert: Assert) {
+        await render(hbs`<HyperTableV2 @handler={{this.handler}} @features={{hash selection=true}} />`);
+        await click('.hypertable__column.hypertable__column--selection header .upf-checkbox');
+
+        assert.dom('.selected-count').hasText(this.handler.selection.length.toString());
+      });
+
+      test('it renders the partial mode for the head checkbox', async function (assert: Assert) {
```

Comment:
the head checkbox ?
It's the one from the header right ?

Maybe `it renders the partial mode on the select-all checkbox` ? 🤷

Reply:
I also change that in the other tests. I rename `header checkbox` to `select-all checkbox` to be more coherent